docs: update olap docs (#6950)
buildbuddy-io/buildbuddy: docs/config-olap-databases.md

sidebar_label: OLAP Database
---

## Introduction

> To speed up the analysis of historical build event data, BuildBuddy relies on ClickHouse, an OLAP database solution.

Reviewer: ```suggestion
To speed up the analysis of historical build event data, BuildBuddy can be configured to use ClickHouse as an OLAP database, in addition to the primary SQL database required for core functionality.
```

Reviewer: I think these suggested edits got reverted somehow

Author: Yeah, it's accidental. I fixed it in the latest version.

Could you please take another look?

> To speed up the analysis of historical build event data, BuildBuddy can be configured to use ClickHouse as an OLAP database, in addition to the primary SQL database required for core functionality.

Reviewer 2: Would be nice to list a few features that depend on clickhouse.
- Historical build events (Trends page)
- Test Grid
- Tags
- Audit Logs

Author: I added tag filtering and audit logging to the list of features depending on OLAP DB.

> Deploying ClickHouse is optional when using BuildBuddy
> BuildBuddy does not require ClickHouse for its core features, such as Build Event Service, Remote Cache, and Remote Execution.

Reviewer: ```suggestion
Setting up ClickHouse is completely optional when using BuildBuddy.
BuildBuddy does not require ClickHouse for its core features, including the build results UI, remote cache, and remote execution system.
```

> However, some UI features, such as Trends, Drilldown, and Test Grid, do rely on ClickHouse.

Reviewer: I think the tests page used to work with MySQL and might still work with it? It just doesn't scale as well. Also Drilldown is part of the trends tab so the wording can be simplified

```suggestion
However, some UI features, such as Trends, do require ClickHouse.
```

> Without a configured ClickHouse instance, these features will either not be displayed on our UI or will operate in a different mode.

Reviewer: ```suggestion
Without a configured ClickHouse instance, these features will either be missing from the UI, or will be missing some features and may not scale to larger amounts of data.
```

## Options

**Optional**

`olap_database:` The OLAP (online analytical processing) database section configures the OLAP database that BuildBuddy uses to enable the Trends page. **Optional**

- `data_source` This is a connection string used by the database driver to connect to the database. ClickHouse database is supported.

> - `enable_data_replication` If ClickHouse is using a [Cluster Deployment](https://clickhouse.com/docs/en/architecture/cluster-deployment), this will enable data replication within the cluster.

Reviewer: ```suggestion
- `enable_data_replication` If ClickHouse is using a [cluster deployment](https://clickhouse.com/docs/en/architecture/cluster-deployment), this will enable data replication within the cluster.
```

## Example sections

> Example single instance clickhouse configuration

Reviewer: ```suggestion
Example single-instance ClickHouse configuration:
```

> ```

Reviewer: ```suggestion
```yaml
```

olap_database:
  data_source: "clickhouse://buildbuddy_user:pAsSwOrD@12.34.56.78:9000/buildbuddy_db"
```

> Example cluster clickhouse configuration

Reviewer: ```suggestion
Example ClickHouse cluster configuration:
```

> ```

Reviewer: ```suggestion
```yaml
```
